.github/workflows/test.yml

```diff
@@ -0,0 +1,34 @@
+name: Ruby CI
+
+on:
+  - push
```

Comment:
Could we limit CI on push to only the main branch? Or is there a reason why it's all branches?

Reply:
I'll take a look. For whatever reason numerous gems in our space are setup this way.

Reply:
https://github.com/Shopify/browser_sniffer/pull/39/commits/88b5420489f55335a1946a7c1fab0e95593addee should do it. I was worried that `pull_request` event was only on PR creation, but there's also a PR event called `synchronize` from github whenever the branch is updated.